lib/helpers/schema-utils.js

```diff
@@ -35,14 +35,20 @@ module.exports.getSchemaByFilesPath = (apiDefinitionsPath, buildSchemaOptions) =
 };
 
 module.exports.getValidatorByPathMethodAndCode = (schema, request, response) => {
-  const route = pathMatcher(schema, request.path);
+  const route = pathMatcher(schema, request.path, request.method);
   return get(schema, `${route}.${request.method}.responses.${response.status}`);
 };
 
 module.exports.pathMatcher = pathMatcher;
-function pathMatcher(routes, path) {
+function pathMatcher(routes, path, method) {
   return Object
     .keys(routes)
+    .sort((firstEl, secondEl) => {
+      const firstResult = firstEl.split(':').map((value, index) => `${index}`).join('');
```

Comment:
this sort doesn't take into account different positions that `:` can be, i.e

```js
> '/aaaa/:bbbb'.split(':').map((value, index) => `${index}`).join('')
'01'
> '/aaaa:/bbbb'.split(':').map((value, index) => `${index}`).join('')
'01'
```
this logic takes into account the path segment that `:` was found in and not its absolute position
```js
> '/aaaa/:bbbb'.split('/').map((seg) => seg.includes(':')).map((seg, i, arr) => seg * Math.pow(10, arr.length - i)).reduce((sum, seg) => sum + seg, 0)
10
> '/aaaa:/bbbb'.split('/').map((seg) => seg.includes(':')).map((seg, i, arr) => seg * Math.pow(10, arr.length - i)).reduce((sum, seg) => sum + seg, 0)
100
```

anyway, the weight calculation should be extracted to a function